### PR TITLE
View all membership requests

### LINKED ISSUE #17: List out the membership requests
### Is your feature request related to a problem?

> Since the admin can take an action [review, accept, reject] each of the membership requests, The admin should be provided with a list of all requests and can choose any from within.

#### :sparkles: Describe the solution you'd like :sparkles:

> For the admin to an action on the membership request, they should be able to get a list of requests that were sent along with their full details.

#### :part_alternation_mark: Describe alternatives you've considered :part_alternation_mark:

> None

#### Additional context

> Each item should be fetched with all the required data for the admin to see what it entails and help them conduct the background checking.

### PR DESCRIPTION
**Pull request by mcaleb808**
---

#### Description of changes made

> Admin can view all the membership requests.

#### Type of change

> **Please select the relevant options**

- [x] New feature (non-breaking changes which add the new functionalities)
- [ ] Bug fix (non-breaking changes which fix an issue)
- [ ] Breaking change (fix or a feature that would cause existing feature not to work as expected)
- [x] This change requires a documentation update

#### How Has This Been Tested?

> **Please describe the tests that you ran to verify your changes**

- [x] Unit
- [ ] Integration
- [ ] End to end

#### How should this be manually tested?

> **Provide the steps to manually test the functionalities implemented in this PR**

- After cloning the [repo](https://github.com/mcaleb808/myPharmBackend.git), and `cd into the project`
- Check out this branch by _git checkout `branchName`_
- Run `yarn | npm install` to install the project dependencies
- Make sure to have your `.env` file set up correctly
- Run `yarn dev | npm run dev` to start the local server.
- using `Postman` send `GET` request to `/api/v1/requests` to view all requests.

#### Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have ensured coding conventions for my code prior to submission.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove the effectiveness of my work.
- [x] new and existing tests pass locally with my changes.

#### GitHub issue your changes fix

Resolves #17

#### Screenshots
N/A

### CODE REVIEW

File: api/v1/services/PharmService.js [hunk 1]
```diff
@@ -9,6 +9,14 @@ export default class PharmService {
     }
   }
 
+  static async getAllRequests() {
+    try {
+      return await Pharmacie.findAll({ where: { status: 'request' } });
```
Comment (Musigwa) [inline api/v1/services/PharmService.js]: What if the list is empty?

File: api/v1/services/PharmService.js [hunk 2]
```diff
@@ -9,6 +9,14 @@ export default class PharmService {
...
+    } catch (err) {
+      throw err;
```
Comment (Musigwa) [inline api/v1/services/PharmService.js]: Are you really catching the errors and respond to the user accordingly?

File: __tests__/index.test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  let app;
+  beforeEach(() => {
+    app = server;
+    return app.close();
+  });
+
+  afterEach(() => app.close());
+  it('test root endpoint', async () => {
+    const res = await request(app).get('/');
+    expect(res.body.message).toBe('Welcome to EasyPharm API');
```
Comment (Musigwa) [inline __tests__/index.test.js]: It's not recommended to strongly check for equality for the string values, you should rather check whether it contains some relevant keywords

File: api/v1/services/PharmService.js (hunk 1 above)
Comment (mcaleb808) [reply api/v1/services/PharmService.js]: This is the response you get.

Screen Shot 2019-12-17 at 10 35 05

File: api/v1/services/PharmService.js (hunk 2 above)
Comment (mcaleb808) [reply api/v1/services/PharmService.js]: @Musigwa errors are handled, please spare some time to test this feature locally and let me know if you face an error.

File: __tests__/index.test.js (hunk 3 above)
Comment (mcaleb808) [reply __tests__/index.test.js]: @Musigwa kindly provide an example of a relevant keyword we can test in this particular scenario. thank you.

**Pull request merged**